### ISSUE TITLE
Improved notifications

### ISSUE THREAD

**Issue by danboid** [solus-project/budgie-desktop#524]
Would it be possible to make budgie respect the -t switch of notify-send? Under MATE and XFCE, they will display the notification for as long as notify-send specifies but Budgie seems to ignore the time switch and only ever show the notification for a few seconds before its relegated to the panels notification archive. 

My second suggestion would be to improve the noticeability  (if thats a word) of the panel notification bell icon when there are unread notifications as I find it all too easy to miss the little red bell on a FHD display or better. Last time I spoke to you about it you (Ikey) said that Budgie would be compositing only so I'd like too see that put to good use here with a red, semi-transparent, ripple-like pulse effect emanating from the notification bell icon that overlaps but doesn't obscure the adjacent icons when there are new notification(s).

Do you think clutter/mutter or whatever would enable this or could it be done with plain GTK3? I think it'd look really cool!

**Comment by ikeydoherty** [comment]
Clutter/Mutter form our window manager, so they're not included in this discussion.
We can add an animation I agree, but it won't be a ripple, that's material design territory.

Budgie enforces a sanity limit on notification timeouts, minimum 4 seconds and maximum of 20 seconds. Are you using miliseconds with the -t switch here? (You should)

**Comment by danboid** [comment]
Yes, I'd imagine Mutter is irrelevant in this discussion but I thought clutter could be used to achieve the sort of animated effect I'm talking about? Maybe ripple is the wrong or too fancy a description? What I envision is Budgie drawing red, semi-transparent, concentric circles that would fade out as they get further (although only a short distance) from the notification icon. Is this not clutter territory?

Yes, I am using milliseconds with notify-send - 7200000 of them to be precise. I have it set so high (a couple of hours) for a couple of my notifications (Skype calls and messages) so that if someone sends me a message just before I go for lunch and then I have a meeting or something straight after lunch I should still see it on my desktop.

**Comment by ikeydoherty** [comment]
It is clutter territory, and the panel is GTK - so its not possible. I also don't want to make Budgie any fatter than it is.

Again:
> Budgie enforces a sanity limit on notification timeouts, minimum 4 seconds and maximum of 20 seconds

720000 > 20 seconds.

**Comment by ikeydoherty** [comment]
MATE forked an entire desktop and XFCE took over 2 years from my prompt to decide that they would finally implement GTK3 and stop pussyfooting around the issue with Qt or GTK, so its hardly a valid example of folks making good decisions.

I explained _why_ your notifications aren't showing up. The solution here is *not* absurdly long time outs, that's just just working around deficiencies. The solution is an *idle timer* to know when the user is active or not, and suspend countdowns when the user is inactive.

**Comment by ikeydoherty** [comment]
For reference, when I say working around deficiencies, I'm talking about Budgie itself. It should already not be stupid, but it is.
